modules/dbparser/groupingby.c

```diff
@@ -277,6 +276,7 @@ grouping_by_expire_entry(TimerWheel *wheel, guint64 now, gpointer user_data)
     }
   else
     {
+      stateful_parser_emit_synthetic(&self->super, msg);
```

Comment:
What's the motivation behind this extraction? 
Is that `grouping_by_emit_synthetic` violates single-responsibility?

Reply:
I don't get the either.

This is not a refactor (extraction) commit as the message suggests, you've removed `if (msg)` and `stateful_parser_emit_synthetic()` is not NULL-safe.